src/core/thread/mle_router.cpp

```diff
@@ -1369,9 +1361,15 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
 
                     break;
                 }
+
+                if (mRouterSelectionJitterTimeout == 0 &&
```

Comment:
here, for REED, does it have to process Advertisement only from its parent to decide whether or not to become an active Router?